client/my-sites/hosting/phpmyadmin-card/restore-db-password.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getHttpData, httpData, requestHttpData } from 'state/data-layer/http-data';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import Dialog from 'components/dialog';
+import { connect } from 'react-redux';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { successNotice } from 'state/notices/actions';
+
+const requestId = siteId => `restore-db-password-${ siteId }`;
+
+export const restoreDatabasePassword = siteId =>
+	requestHttpData(
+		requestId( siteId ),
+		http(
+			{
+				method: 'POST',
+				path: `/sites/${ siteId }/hosting/restore-database-password`,
+				apiNamespace: 'wpcom/v2',
+				body: {},
+			},
+			{}
+		),
+		{
+			fromApi: () => () => [ [ requestId( siteId ), null ] ],
+			freshness: 0,
+		}
+	);
+
+const RestorePasswordDialog = ( {
+	isVisible,
+	onRestore,
+	onCancel,
+	isRestoring,
+	hasRestored,
+	siteId,
+	translate,
+	showSuccessNotice,
+	reset,
+} ) => {
+	if ( hasRestored ) {
+		reset();
+		showSuccessNotice( translate( 'Your database password has been restored.' ), {
+			duration: 3000,
+		} );
+		onRestore();
+	}
+
+	const buttons = [
+		{
+			action: 'restore',
+			label: translate( 'Restore' ),
+			onClick: () => restoreDatabasePassword( siteId ),
```

Comment:
If you're looking for request information, I `restoreDatabasePassword( siteId )` should also return an object.